### ISSUE TITLE
FNX-5106 ⁃ Handle Play Store links

### ISSUE THREAD

**Issue by AaronMT** [mozilla-mobile/fenix#1570]
## User Story
* As a user, when I click on a link to download/open an app, I want my browser to give me the option to download the app from the Google Play Store or open the app if I already have it installed, so I can easily choose to download/access an app.

### Acceptance Criteria
-I can choose to download an app when I click on a download/open link on a website
-I can choose to open an installed app when I click on a download/open link on a website

Examples:

* Visit http://imgur.com
* Tap 'Get App'

**Comment by AaronMT** [comment]
Another, e.g, http://newegg.ca, tap View on the top banner

**Comment by ekager** [comment]
This also currently does not work in reference browser. We could probably add some logic dealing with these external URIs and launch an intent in `AppRequestInterceptor` https://github.com/mozilla-mobile/fenix/blob/master/app/src/main/java/org/mozilla/fenix/AppRequestInterceptor.kt

But there is also an issue open in AC about this issue: https://github.com/mozilla-mobile/android-components/issues/1132

And one open in Reference Browser: https://github.com/mozilla-mobile/reference-browser/issues/279

Because most browsers will want this functionality, maybe this is something that should be in AC.

**Comment by pwd-github** [comment]
Any chance this could get a more meaningful title so that it's easier to locate? Handle Play Store links is incredibly specific.

**Comment by vesta0** [comment]
Blocked on https://github.com/mozilla-mobile/android-components/issues/1132

**Comment by bifleming** [comment]
@kbrosnan does this need to go through QA?

**Comment by andreicristianpetcu** [comment]
related https://github.com/mozilla-mobile/fenix/issues/3058#issuecomment-497832247

**Comment by sv-ohorvath** [comment]
Not completely fixed: 
If the app is not installed, the link doesn't work: PS not launched.
If the app is already installed it will launch the app.
build: 6/4 (#11550608)

**Comment by AaronMT** [comment]
We should probably just treat that as a new issue.

**Comment by vesta0** [comment]
I created an issue for the scenario reported above: #3177 
Here is another scenario related to phone intents #3022 

@kbrosnan @sv-ohorvath would it be possible to run a few different scenarios to identify Intent use cases that need to be built/fixed and create new bugs for them?

Once we have logged all the major failed scenarios, we can close this issue.

**Comment by sv-ohorvath** [comment]
Logged a new one: #3205 and added a test page with some deep links on #3177. I'll investigate more and log new issues as I'll find them.
@kbrosnan If you have anything to add before we close this...

**Comment by vesta0** [comment]
FYI created a Meta #3223 to track all individual related issues.

**Comment by sv-ohorvath** [comment]
I think we can close this.

**Comment by kbrosnan** [comment]
This was backed out in PR #3359

**Comment by vesta0** [comment]
Will be addressed in #3177

**Comment by karlcow** [comment]
@ekager will this bug cover also the currently unknown schemes such as 

```html
<a href="mailto:foo@example.com">Bariton</a>
```

Because of a couple of issues have been closed against this one and this bug doesn't make it clear if it's only `intent://` scheme or any scheme. 
Thanks.
